TWLight/applications/views.py

```diff
@@ -841,19 +886,55 @@ def post(self, request, *args, **kwargs):
         # important work for Applications. This includes handling the closing
         # dates for applications and sending email notifications to editors
         # about their applications.
+        batch_update_failed = []
+        batch_update_success = []
         for app_pk in request.POST.getlist('applications'):
             try:
                 app = Application.objects.get(pk=app_pk)
             except Application.DoesNotExist:
                 logger.exception('Could not find app with posted pk {pk}; '
                     'continuing through remaining apps'.format(pk=app_pk))
                 continue
-
-            app.status = status
-            app.save()
-        #Translators: After a coordinator has changed the status of a number of applications, this message appears.
-        messages.add_message(request, messages.SUCCESS,
-            _('Batch update successful.'))
+            if app.partner.authorization_method == Partner.PROXY:
+                total_accounts_available_for_distribution = None
+                if app.specific_stream is not None:
+                    if app.specific_stream.accounts_available is not None:
+                        total_apps_approved_or_sent = User.objects.filter(
+                                              editor__applications__partner=app.partner,
+                                              editor__applications__status__in=(Application.APPROVED, Application.SENT),
+                                              editor__applications__specific_stream=app.specific_stream).count()
+                        total_accounts_available = app.specific_stream.accounts_available
+                        total_accounts_available_for_distribution = total_accounts_available - total_apps_approved_or_sent
+                elif app.partner.accounts_available is not None:
+                    total_apps_approved_or_sent = User.objects.filter(
+                                          editor__applications__partner=app.partner,
+                                          editor__applications__status__in=(Application.APPROVED, Application.SENT),
+                                          editor__applications__specific_stream=None).count()
+                    total_accounts_available_for_distribution = app.partner.accounts_available - total_apps_approved_or_sent
+
+                if int(status) == Application.APPROVED:
+                    try:
+                        assert app.partner.status != Partner.WAITLIST and (total_accounts_available_for_distribution > 0 if total_accounts_available_for_distribution else True)
+                        batch_update_success.append(app_pk)
```

Comment:
This is a nice way of approving as many as possible before hitting the limit, but I wonder if we'd be better denying the batch outright so that the coordinator can decide which of the applications they want to approve, so that they can prioritise. I'll ask Nikki on Slack.

Reply:
Per Nikki's feedback on Slack, it would be preferable to take no actions so that the coordinator can decide which applications to approve.